config.go

```diff
@@ -641,12 +641,14 @@ func loadConfig(ctx context.Context) (*config, []string, error) {
 
 	// Set dialer and DNS lookup functions if proxy settings are provided.
 	if cfg.Proxy != "" {
-		proxy := &socks.Proxy{
+		proxy := socks.Proxy{
 			Addr:         cfg.Proxy,
 			Username:     cfg.ProxyUser,
 			Password:     cfg.ProxyPass,
 			TorIsolation: cfg.TorIsolation,
 		}
+		proxyPool := socks.NewPool(proxy, 32, 128)
```

Comment:
I would prefer if there was no queuing, turning this into a hard cap with errors returned immediately if a new request would otherwise exceed the cap.

Feel free to remove the queuing code from go-socks if you don't think anything else will be using it.

We can also make this a --proxy.circuitlimit flag with default 32.